contracts/StrategyCurveTwoPool.sol

```diff
@@ -0,0 +1,383 @@
+
+    // Arbitrum healthcheck contract address
+    healthCheck = _contractsAddress.healthCheck;
+
+    // Our standard approvals. want = Curve LP token
+        want.approve(_contractsAddress.gauge, type(uint256).max);
+        crv.approve(_contractsAddress.router, type(uint256).max);
+
+    maxReportDelay = 2 days; // 2 days in seconds
+  }
+
+  /* ========== VIEWS ========== */
+
+  function stakedBalance() public view returns (uint256) {
+    return gauge.balanceOf(address(this));
+  }
+
+  function balanceOfWant() public view returns (uint256) {
+    return want.balanceOf(address(this));
+  }
+
+  function estimatedTotalAssets() public view override returns (uint256) {
+    return balanceOfWant().add(stakedBalance());
+  }
+
+  /* ========== MUTATIVE FUNCTIONS ========== */
+  // these should stay the same across different wants.
+
+  function adjustPosition(uint256 _debtOutstanding) internal override {
+    if (emergencyExit) {
+      return;
+    }
+    // Send all of our LP tokens to deposit to the gauge if we have any
+    uint256 _toInvest = balanceOfWant();
+    if (_toInvest > 0) {
+      gauge.deposit(_toInvest);
+    }
+  }
+
+  function liquidatePosition(uint256 _amountNeeded)
+    internal
+    override
+    returns (uint256 _liquidatedAmount, uint256 _loss)
+  {
+    uint256 _wantBal = balanceOfWant();
+    if (_amountNeeded > _wantBal) {
+      // check if we have enough free funds to cover the withdrawal
+      uint256 _stakedBal = stakedBalance();
+      if (_stakedBal > 0) {
+        gauge.withdraw(Math.min(_stakedBal, _amountNeeded.sub(_wantBal)));
+      }
+      uint256 _withdrawnBal = balanceOfWant();
+      _liquidatedAmount = Math.min(_amountNeeded, _withdrawnBal);
+      _loss = _amountNeeded.sub(_liquidatedAmount);
+    } else {
+      // we have enough balance to cover the liquidation available
+      return (_amountNeeded, 0);
+    }
+  }
+
+  function liquidateAllPositions() internal override returns (uint256) {
+    uint256 _stakedBal = stakedBalance();
+    if (_stakedBal > 0) {
+      // don't bother withdrawing zero
+      gauge.withdraw(_stakedBal);
+    }
+    return balanceOfWant();
+  }
+
+    function _claimRewards() internal {
+        gaugeFactory.mint(address(gauge));
+    }
+
+    function claimRewards() external onlyVaultManagers {
+        // Claims any pending CRV
+        //
+        // Mints claimable CRV from the factory gauge. Reward tokens are sent to `msg.sender`
+        // The method claim_rewards() from the old gauge now only applies to third-party tokens.
+        // There are no third-party tokens in this strategy.
+        _claimRewards();
+    }
+
+  function prepareMigration(address _newStrategy) internal override {
+        // Withdraw LP tokens from the gauge. The transfer to the new strategy is done
+        // by migrate() in BaseStrategy.sol. Note that this function does not claim
+        // any pending rewards. To claim and transfer pending rewards to a new
```

Comment:
Can probably delete the last sentence of this comment

Reply:
will do